### DISCUSSION TITLE
Any great RPC implementation?

### DISCUSSION THREAD

**Discussion by hldup** [gftea/amqprs#144] [Q&A]
I'm still a newbie when it comes to AMQP but for my usecase I need rpc. I've seen some other libraries that had great examples but the library was either not maintained or the code was overcomplicated.

**Comment by michaelklishin** [comment]
I don't know what you consider to be "overcomplicated" but tutorial 6 ("request-response") is available for half a dozen programming languages and can be copied pretty directly to Rust https://www.rabbitmq.com/tutorials#6-rpc.

A more efficient version that will take a bit more effort to understand at first would use [Direct Reply-to](https://www.rabbitmq.com/docs/direct-reply-to) which does not require any special client library support.
